### DISCUSSION TITLE
Course Progress Exhaust to capture number of attempts

### DISCUSSION THREAD

**Discussion by mmanimegalai** [Sunbird-Ed/Community#37] [Enhancement request]
This is a DIKSHA request

Currently, if there is a Course assessment attached to a Course, the Course Progress Exhaust captures the best attempt score (%) of the course assessment for each user.

The information about in which attempt did the user first passed the required criteria and got the certificate is a very important information. This is useful to get metrics about % of users who are passing the assessment within first or second attempts which will help derive insights like - how difficult or easy the assessment is. Hence this information should be included to the existing course progress exhaust

**Comment by sharathkashyap** [comment]
@mmanimegalai I believe this information is already present with us in the form of API response. We would just need a corresponding visualisation to be done in SunbirdED to represent the information to the user as well. In reports as well this information can be presented. This would require some extension to be done in Observ and Lern.
cc: @luckynvdu @rhwarrier

**Reply by mmanimegalai** [reply]
@rhwarrier @reshmi-nair

**Reply by rhwarrier** [reply]
Lern will take this up - you can follow progress here : https://project-sunbird.atlassian.net/browse/LR-127
